service/common/src/main/java/org/apache/polaris/service/events/listeners/FileBufferPolarisEventListener.java

```diff
@@ -0,0 +1,150 @@
+
+        for (int i = 0; i < bufferShardCount; i++) {
+            buffers.put(i, createShard(i));
+        }
+    }
+
+    @Override
+    void addToBuffer(PolarisEvent polarisEvent, CallContext callCtx) {
+        long ingestionTime = System.currentTimeMillis();
+        int shardNum = polarisEvent.hashCode() % bufferShardCount;
+        BufferShard bufferShard = buffers.get(shardNum);
+        if (bufferShard == null) {
+            LOGGER.error("No buffer shard found for shard #{}. Event dropped: {}", shardNum, polarisEvent);
+            return;
+        }
+        if (bufferShard.polarisEvent == null) {
+            bufferShard.polarisEvent = polarisEvent;
+        }
+        try {
+            ObjectOutputStream oos = bufferShard.oos;
+            oos.writeObject(polarisEvent);
+            oos.flush();
+            oos.reset();
+        } catch (IOException e) {
+            LOGGER.error("Unable to write event: {} to buffer shard: {}", polarisEvent, e);
+        }
+
+        if (ingestionTime - bufferShard.polarisEvent.getTimestampMs() > this.timeToFlush) {
+            // Create a new shard (file and writer)
+            buffers.put(shardNum, createShard(shardNum));
+            // Create and fire task to persist the events in the old shard file
+            PolarisMetaStoreManager polarisMetaStoreManager = metaStoreManagerFactory.getOrCreateMetaStoreManager(callCtx.getRealmContext());
+            PolarisBaseEntity taskEntity =
+                    new PolarisEntity.Builder()
+                            .setId(polarisMetaStoreManager.generateNewEntityId(callCtx.getPolarisCallContext()).getId())
+                            .setCatalogId(0L)
+                            .setName("persist_logs_and_cleanup_" + bufferShard.shardFile.getName())
+                            .setType(PolarisEntityType.TASK)
+                            .setSubType(PolarisEntitySubType.NULL_SUBTYPE)
+                            .setCreateTimestamp(System.currentTimeMillis())
+                            .build();
+            Map<String, String> properties = new HashMap<>();
+            properties.put(
+                    PolarisTaskConstants.TASK_TYPE,
+                    String.valueOf(AsyncTaskType.LOG_PERSISTENCE_AND_CLEANUP.typeCode()));
+            properties.put("file", bufferShard.shardFile.getAbsolutePath());
+            taskEntity.setPropertiesAsMap(properties);
+            EntityResult entityResult = polarisMetaStoreManager.createEntityIfNotExists(callCtx.getPolarisCallContext(), null, taskEntity);
+            taskExecutor.addTaskHandlerContext(entityResult.getEntity().getId(), callCtx);
+        }
+    }
+
+    private BufferShard createShard(int shardNum) {
+        File file = new File(System.getProperty("java.io.tmpdir") + "/polaris-event-buffer-shard-" + shardNum + "file_timestamp-" + System.currentTimeMillis());
```

Comment:
This needs to be customizable, as well as the number of buffers

Reply:
The file location could matter; imagine someone wants to use different disks for different shards